spec/controllers/bales_controller_spec.rb

```diff
@@ -198,4 +198,51 @@ def update_bale_call(id, weight, material)
       end
     end
   end
+  describe 'GET # show_bales_by_material' do
+    let!(:device) do
+      create(:device, device_id: '1', device_type: 'android',
+                      organization: organization)
+    end
+    let!(:auth_user) { create_an_authenticated_user_with(organization, device.device_id, device.device_type) }
+    let!(:bale) { create(:bale, organization: organization, user: auth_user) }
+
+    def show_bales_by_material_call(material)
+      @request.headers['material'] = material
+      get :show_by_material
+    end
+
+    context 'when user is authenticated' do
+      it 'does return success' do
+        show_bales_by_material_call(%w[Plastic Trash Glass].sample)
+        expect(response).to have_http_status(:ok)
+      end
+
+      it 'does return the bales' do
+        Bale.first.update(material: 'Trash')
+        show_bales_by_material_call('Trash')
+        expect(json_response.count).to eql 1
+      end
+    end
+    context 'when material is incorrect or missing' do
+      it 'does return the right error (invalid)' do
```

Comment:
try to be more descriptive with test description, i mean, if this test fail and i try to understand what is checking, it is impossible to do it with this description. Maybe something like `does return an error when material is invalid`. I know that it isn't a big change but it would be worth in the near future.